src/modm/platform/adc/stm32f3/adc_impl.hpp.in

```diff
@@ -410,3 +434,70 @@ modm::platform::Adc{{ id }}::acknowledgeInterruptFlags(const InterruptFlag_t fla
 	// Writing a zero is ignored.
 	ADC{{ id }}->ISR = flags.value;
 }
+
+/**
+ * @arg slot for the offset register (0..3)
+ * @arg channel channel to which the offset is applied
+ * @arg offset offset value to be applied to the channel
+ * @return true if configuration is successful, false if the ADC is currently converting
+ */
+bool
+modm::platform::Adc{{ id }}::setChannelOffset(const OffsetSlot slot, const Channel channel, const int16_t offset,
+	const bool saturate, const bool enable)
+{
+	if ( (ADC{{ id }}->CR & ADC_CR_JADSTART) || (ADC{{ id }}->CR & ADC_CR_ADSTART) ) {
+		// ADC is currently converting, cannot set offset
+		return false;
+	}
+
+%% if target["family"] in["f3", "l4", "l5"] or (target["family"] in["h7"] and target["name"][0] in["4", "5", "a", "b"]):
+	// F3 or H74x/H75x does not support signed offsets, so signMask is always 0
+	if (offset < 0) {
+		return false;	// F3 ADC offsets must be positive
+	}
+	const uint32_t signMask = 0u;
+
+	// F3 or H74x/H75x does not support saturation, so saturateMask is always 0
+	if (saturate) {
+		return false;	// F3 ADC does not support saturation
+	}
+	const uint32_t saturateMask = 0u;
+
+%% elif target["family"] in["h7"]
+	// H7 names the bits differently, but the logic is the same
```

Comment:
```suggestion
%# H7 names the bits differently, but the logic is the same
```
jinja2 comments is better as they won't show up in the generated code where this comment makes no sense without context.